lib/puppet/provider/radius_server/cisco_nexus.rb

```diff
@@ -0,0 +1,108 @@
+require 'puppet/resource_api/simple_provider'
+require 'puppet_x/cisco/cmnutils'
+
+# Implementation for the radius_server type using the Resource API.
+class Puppet::Provider::RadiusServer::CiscoNexus < Puppet::ResourceApi::SimpleProvider
+  def canonicalize(_context, resources)
+    resources
+  end
+
+  RADIUS_SERVER_PROPS = {
+    auth_port:           :auth_port,
+    acct_port:           :acct_port,
+    timeout:             :timeout,
+    retransmit_count:    :retransmit_count,
+    accounting_only:     :accounting,
+    authentication_only: :authentication,
+  }
+
+  UNSUPPORTED_PROPS = [:group, :deadtime, :vrf, :source_interface]
+
+  def get(context, _names=nil)
+    require 'cisco_node_utils'
+
+    radius_servers = []
+    Cisco::RadiusServer.radiusservers.each_value do |v|
+      radius_servers << {
+        ensure:               'present',
+        name:                 v.name,
+        auth_port:            v.auth_port ? v.auth_port : nil,
+        acct_port:            v.acct_port ? v.acct_port : nil,
+        timeout:              v.timeout ? v.timeout : -1,
+        retransmit_count:     v.retransmit_count ? v.retransmit_count : -1,
+        key:                  v.key ? v.key : 'unset',
+        key_format:           v.key_format ? v.key_format : -1,
+        accounting_only:      v.accounting,
+        authentication_only:  v.authentication
+      }
+    end
+
+    PuppetX::Cisco::Utils.enforce_simple_types(context, radius_servers)
+  end
+
+  def munge_flush(val)
+    if val.is_a?(String) && val.eql?('unset')
+      nil
+    elsif val.is_a?(Integer) && val.eql?(-1)
+      nil
+    elsif val.is_a?(Symbol) && val.eql?(:true)
+      true
+    elsif val.is_a?(Symbol) && val.eql?(:false)
```

Comment:
Don't think the value would ever be `:true` or `:false` as it would be a syntax error anyway and the catalog wouldn't compile; something like:

```
Error: Could not parse for environment production: Syntax error at ':' (file: /Users/thomas.franklin/puppet/cisco-network-puppet-module/test.pp, line: 6, column: 28) on node nexus
```

Reply:
Copypasta from the old provider, now removed this from validation.